File: benchmarks/src/main/clojure/filter-map-reduce.clj
(ns bench)

(defn mod500 [n] (mod n 500))

(defn lazy []
  (reduce + 0
    (filter even?
      (map inc
        (map mod500
          (take 2000000 (iterate inc 0)) )))))

(defn run [data]
  (reduce + 0
    (filter even?
      (map inc data) )))

File: benchmarks/src/main/js/filter-map-reduce.js
var data = new Array(2000000);
for (var i = 0; i < 2000000; i++) {
  data[i] = i % 500;
}

function run() {
  return data.map(function(x) {
    return x + 2;
  }).filter(function(x) {
    return x % 2 == 0;
  }).reduce(function(acc, x) {
    return acc + x;
  }, 0);
}